homeassistant/components/cover/tahoma.py

```diff
@@ -27,27 +36,84 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 class TahomaCover(TahomaDevice, CoverDevice):
     """Representation a Tahoma Cover."""
 
+    def __init__(self, tahoma_device, controller):
+        """Initialize the device."""
+        super().__init__(tahoma_device, controller)
+
+        self._closure = 0
+        # 100 equals open
+        self._position = 100
+        self._closed = False
+        self._icon = None
+        # Can be 0 and bigger
+        self._lock_timer = 0
+        # Can be 'comfortLevel1'
+        self._lock_level = None
+        # Can be 'wind'
+        self._lock_originator = None
+
     def update(self):
         """Update method."""
         self.controller.get_states([self.tahoma_device])
 
-    @property
-    def current_cover_position(self):
-        """
-        Return current position of cover.
+        if 'core:ClosureState' in self.tahoma_device.active_states:
+            self._closure = \
+                self.tahoma_device.active_states['core:ClosureState']
+        else:
+            self._closure = None
+        if 'core:PriorityLockTimerState' in self.tahoma_device.active_states:
+            self._lock_timer = \
+                self.tahoma_device.active_states['core:PriorityLockTimerState']
+        else:
+            self._lock_timer = None
+        if 'io:PriorityLockLevelState' in self.tahoma_device.active_states:
+            self._lock_level = \
+                self.tahoma_device.active_states['io:PriorityLockLevelState']
+        else:
+            self._lock_level = None
+        if 'io:PriorityLockOriginatorState' in \
+                self.tahoma_device.active_states:
+            self._lock_originator = \
+                self.tahoma_device.active_states[
+                    'io:PriorityLockOriginatorState']
+        else:
+            self._lock_originator = None
+
+        # Define which icon to use
+        if self._lock_timer > 0:
+            if self._lock_originator == 'wind':
+                self._icon = 'mdi:weather-windy'
+            else:
+                self._icon = 'mdi:lock-alert'
+        else:
+            self._icon = None
 
-        0 is closed, 100 is fully open.
-        """
-        try:
-            position = 100 - \
+        # Define current position.
+        #   _position: 0 is closed, 100 is fully open.
+        #   'core:ClosureState': 100 is closed, 0 is fully open.
+        if 'core:ClosureState' in self.tahoma_device.active_states:
+            self._position = 100 - \
                 self.tahoma_device.active_states['core:ClosureState']
-            if position <= 5:
-                return 0
-            if position >= 95:
-                return 100
-            return position
-        except KeyError:
-            return None
+            if self._position <= 5:
+                self._position = 0
```

Comment:
Why is this?

Reply:
Actually, I don't know. It was there since the beginning (#10652). So, I kept the code.